### PR TITLE
Support sticky footer for the Modal component.

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

https://www.notion.so/joinzoe/COVID-Test-Screen-UI-update-Modals-for-mechanism-and-antibody-type-to-have-sticky-buttons-at-the-b-32c7c41d2150426080b3069d56cc6893

## On what platform have you tested the change?

- [x] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Simulator Screen Shot - iPhone 5s - 2021-07-27 at 12 54 08

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

File: src/routes/index.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default function MainNavigator() {
       <Stack.Screen component={screens.CovidTestDetailScreen} name="CovidTestDetail" options={noHeader} />
       <Stack.Screen component={screens.CovidTestListScreen} name="CovidTestList" options={noHeader} />
       <Stack.Screen component={screens.CreateProfileScreen} name="CreateProfile" options={noHeader} />
-      <Stack.Screen component={screens.DashboardScreen} name="Dashboard" options={noHeader} />
```
Comment (juliatan) [inline src/routes/index.tsx:43]: I think this needs to be reverted.

File: src/components/Modal.tsx [hunk 2]
```diff
@@ -1,45 +1,97 @@
...
+
+// These magic numbers make sure the scroll indicator aligns pixel perfect to the edge on iOS.
+const INSETS = {
+  bottom: -3,
+  left: 0,
+  right: -3,
+  top: -3,
+};
+
+const colors = ['#ffffff00', 'white'];
```
Comment (juliatan) [inline src/components/Modal.tsx:26]: Should we link to `colors`?

File: src/components/Modal.tsx [hunk 3]
```diff
@@ -1,45 +1,97 @@
...
+    backgroundColor: '#ffffff00',
+    height: CONTENT_SPACING,
+    left: CONTENT_SPACING,
+    position: 'absolute',
+    right: CONTENT_SPACING,
+    top: -CONTENT_SPACING,
+    zIndex: 1,
   },
-  modal: {
+  padding: {
```
Comment (juliatan) [inline src/components/Modal.tsx:72]: I don't know how difficult it is but I was comparing the iPhone 11 and iPhone SE. Should there be less spacing between the OK button and end of content on iPhone 11? Looks perfect on SE though. (1st screenshot is iPhone 11)
Simulator Screen Shot - iPhone 11 - 2021-07-27 at 12 23 11
Simulator Screen Shot - iPhone SE (2nd generation) - 2021-07-27 at 12 34 26

File: src/components/Modal.tsx [hunk 4]
```diff
@@ -1,45 +1,97 @@
...
   },
   scrollView: {
-    borderRadius: 16,
+    marginRight: SCROLL_INDICATOR_OFFSET,
+    marginTop: SCROLL_INDICATOR_OFFSET,
+  },
+  view: {
+    flex: 0,
     flexGrow: 0,
+    flexShrink: 1,
```
Comment (juliatan) [inline src/components/Modal.tsx:85]: Damn... I thought `flex: 0` was equivalent to `flexShrink: 1`. 😓

**Review by juliatan — APPROVED** [review]
This is great!! 🙌🏻

File: src/components/Modal.tsx (hunk 3 above)
Comment (aikewoody) [reply src/components/Modal.tsx:72]: Let me have a look

File: src/components/Modal.tsx (hunk 2 above)
Comment (aikewoody) [reply src/components/Modal.tsx:26]: Because these are highly specific colors I don't think that's necessary

**Pull request merged**